### PR TITLE
feat: Make possible to provide image as a secret

### LINKED ISSUE #2917: [feature] generator_container_slsa3 should support passing image as a secret
**Is your feature request related to a problem? Please describe.**

The `image` input of the generator_container_slsa3 workflow is declared as input which prohibits using secrets as values. The following example fails:

```yaml
  provenance:
    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.4.0
    with:
      image: ${{ secrets.REGISTRY_IMAGE }}
      # Passing secrets in any form is impossible in a reusable workflow.
      # Encrypted value could not be decrypteded as the workflow can't be used as a step,
      # and secret outputs are always redacted when are passed between jobs as an output.
    secrets:
      registry-username: ${{ secrets.REGISTRY_USERNAME }}
      registry-password: ${{ secrets.REGISTRY_PASSWORD }}
```

**Describe the solution you'd like**

The `image` input should be available to provide a secret. Example workflow could look liks this:

```yaml
  provenance:
    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.4.0
    with:
      digest: ${{ needs.build.outputs.digest }}
    secrets:
      secret: ${{ secrets.REGISTRY_IMAGE }}
      registry-username: ${{ secrets.REGISTRY_USERNAME }}
      registry-password: ${{ secrets.REGISTRY_PASSWORD }}
```

**Describe alternatives you've considered**

Unfortunately due to destination registry restrictions, we have no alternatives, but to provide every value as a secret, and we are unable to change the overlapping values between the registry endpoint and the published image name.

**Additional context**

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Add option to provide image as a secret for private registries.

Fixes #2917

### CODE REVIEW

**Comment by ianlewis** [comment]
@Danil-Grigorev Thanks!

A couple things:
1. Could you add this to the documentation under the ["Workflow Inputs" heading](https://github.com/slsa-framework/slsa-github-generator/tree/main/internal/builders/container#workflow-inputs)? Be sure to mention that the secret `image` is used over the `image` input if both are provided.
2. Could you add an entry to the [CHANGELOG.md](https://github.com/slsa-framework/slsa-github-generator/blob/main/CHANGELOG.md) under a new "Unreleased" header?

**Comment by Danil-Grigorev** [comment]
Hi @ianlewis, I've updated changelog and workflow inputs as requested, hope it looks alright.

File: CHANGELOG.md [hunk 1]
```diff
@@ -92,6 +95,15 @@ Information on the next release will be added here.
...
 -->
+## Unreleased
+
+This section includes upcoming changes which are not included in the latest release.
+
+### Changes
+
+#### Container generator
+
+- **Fixed**: Passing an image name to the `generator_container_salsa3.yml` containing secret values. (See [#2917](https://github.com/slsa-framework/slsa-github-generator/issues/2917))
```
Comment (ianlewis) [inline CHANGELOG.md:106]:
```suggestion
- **Added**: Passing an image name to the `generator_container_salsa3.yml` containing secret values. (See [#2917](https://github.com/slsa-framework/slsa-github-generator/issues/2917))
```

**Review by ianlewis — APPROVED** [review]

**Comment by ianlewis** [comment]
> Hi @ianlewis, I've updated changelog and workflow inputs as requested, hope it looks alright.

Thanks again for doing this.

**Pull request merged**